src/core/execute-serialize.c

```diff
@@ -1620,6 +1624,15 @@ static int exec_parameters_deserialize(ExecParameters *p, FILE *f, FDSet *fds) {
                                 continue;
 
                         p->exec_fd = fd;
+                } else if ((val = startswith(l, "exec-parameters-handoff-timestamp-fd="))) {
+                        int fd;
+
+                        fd = deserialize_fd(fds, val);
+                        if (fd < 0)
+                                continue;
+
+                        safe_close(p->handoff_timestamp_fd);
+                        p->handoff_timestamp_fd = fd;
```

Comment:
close_and_replace? But for other fds we just do a simple assignment, so be consistent here?

Reply:
i am pretty sure we should rather fix the other invocations to be safe on this. Even if hopefully serialization doesn't contain the same field like this twice, most of the usual "daemon-reload" guards against leaks if this happens, and here we really should too.